config/util.go

```diff
@@ -147,7 +149,21 @@ func DefaultJSONMarshalWithoutHiddenFields(cfg interface{}) ([]byte, error) {
 		if hidden {
 			f.Type = hiddenFieldT
 		}
-		finalStructFields = append(finalStructFields, f)
+
+		// remove omitempty from tag
+		newF := reflect.StructField{
+			Name:      f.Name,
+			PkgPath:   f.PkgPath,
+			Type:      f.Type,
+			Offset:    f.Offset,
+			Index:     f.Index,
+			Anonymous: f.Anonymous,
+		}
+		tagStr := strings.Replace(string(f.Tag), "omitempty", "", 1)
+		tagStr = strings.Replace(tagStr, ",\"", "\"", 1)
```

Comment:
What if its `json:"something,omitempty,somethingelse"`. Does it all break to have a tag like `json:"name," or is the need for this Replace an assumption?